src/services/search/embedding_cache.py

```
@@ -31,9 +31,12 @@
 from pathlib import Path
 from typing import Any
 
-import numpy as np
+try:
+    import numpy as np  # pyre-ignore[21]: optional dep; absent when search extra not installed
+    from numpy.typing import NDArray  # pyre-ignore[21]
+except ImportError as exc:  # pragma: no cover
+    raise ImportError("Install with: pip install 'fo-core[search]'") from exc
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)